### ISSUE TITLE
Stop using feature flags for testing

### ISSUE THREAD

**Issue by Patrik-Stas** [hyperledger/aries-vcx#799]
Per @bobozaur's comment
https://github.com/hyperledger/aries-vcx/pull/797#pullrequestreview-1379809296

> Same thing I mentioned on one of @gmulhearn 's PRs - I think we should aim to remove these feature flags. Not now, as it would be too big a change, but down the line if we all agree on it.
> 
> The way I see it, we should have unit tests which should be fairly self-contained (and can use dev-dependencies) and integration tests which would be more extensive and maybe even require some mocks or other valid services altogether (and can also use dev-dependencies).
> 
> The mocks could then be placed into one/more common integration tests file/s. We could annotate some of the heavier tests with ignore and make use of naming patterns to run certain suites of tests (e.g all tests that start with test_pool_agency).
>
>The rationale is that these feature flags get in the way of code being compiled in IDEs so you don't really see that some change you made messes up some tests (at least not until you run them) and you might end up having made many changes which you then have to accommodate all tests to at once, rather than incrementally.
>
>The feature flags also kind of 'pollute' Cargo.toml, if you ask me.

**Comment by bobozaur** [comment]
> Good idea with naming patterns, I don't quite see a case for using `ignore`  

`ignore` would be used to disable heavy integration tests unless explicitly specified. It's not like we *have to*, but it would be nice to have as implicit behavior running the lighter tests only and to opt-in for heavier stuff, either locally or in CI.

> Mocks are typically used by unit tests though; but possibly some integration tests as well. So I am not sure about this suggestion.

I think that we should ideally avoid creating unit tests that require mocks/external stuff. If something can't be tested as a singular unit, making it an integration test would probably be better. So this way mocks would only be used in one consolidated place.

> Will the proposed changed make it possible to work more incrementally? How?

Removing all these feature flags would allow everyone to immediately see when a change they make breaks some tests, without having to go through the `commit + push + CI` dance or setting up additional services to run tests locally. They would be visible as compile errors if you merely try to test and I believe most IDE's typically run `cargo check` or something so even manually running `cargo test` would be avoided because the IDE would report the errors.

Right now if I were to make a change to some public API function's arguments, I don't know if all the tests are going to work unless I run them (which means running and compiling unrelated tests as well). If I make 10 changes like this before I commit or anything and just let the CI run, then I'll most likely get to see a lot of errors from each individual change and I'd have to fix all of them at once, potentially needing to roll back some stuff.

**Comment by Patrik-Stas** [comment]
Mostly addressed as part of https://github.com/hyperledger/aries-vcx/pull/787

@bobozaur do you think we can close this?

**Comment by bobozaur** [comment]
Well... Not really. They are being used across crates like `libvcx` still. And the `vdrtools` and `modular_libs` are still a huge mess in `aries_vcx` too.

**Comment by Patrik-Stas** [comment]
While feature flags exist to select underlying implementation of components, not used anymore for test categorization. Closing
